### DISCUSSION TITLE
[turborepo] turbo gen. Add support for Github enterprise templates

### DISCUSSION THREAD

**Discussion by HacknBashe** [vercel/turborepo#7463] [Ideas]
### What version of Turborepo are you using?

1.10.7

### What package manager are you using / does the bug impact?

Yarn v2/v3 (node_modules linker only)

### What operating system are you using?

Mac

### Describe the Bug

When running the command `yarn turbo gen workspace --copy` and a url pointing to our enterprise github, I get the error.

```
>>> Invalid URL: "https://github.mycompany.com/shared/templates-monorepo.git". Only GitHub repositories are supported. Please use a GitHub URL and try again.
```

There is line [here](https://github.com/vercel/turbo/blob/a2b1fd47d45d4d5145ed2141481a3136e7ea1576/packages/turbo-utils/src/createProject.ts#L58) that filters out any urls that don't match github.com but I'm not sure if there's more that needs to be changed in order for this to work. I tried changing it locally and testing it but I was struggling to run my local version since it is invoked by the binary through npx.

### Expected Behavior

I would expect to be able to run turbo gen copy against our enterprise github urls.

### To Reproduce

Run the command `yarn turbo gen workspace --copy` in a repo with your own hosted github repo as the value for the copy flag.

**Comment by chris-olszewski** [comment]
Hi @NickHackford, could you provide an example URL of the enterprise repo you're trying to use (doesn't have to be exact)? Also, do you know if there's any authentication required to access the template?

**Comment by HacknBashe** [comment]
Hey Chris. The url our github is hosted at is https://github.ourcompany.com and I cannot see the template unless I am logged into my enterprise github account.

**Comment by tknickman** [comment]
Hey @NickHackford, sorry for the delay here, we should be able to make this work, but it is _slightly_ more involved than the line you pointed out! We have the check that you found in place because all of our utilities to actually download the files we need are specific to Github. You can check those out here: https://github.com/vercel/turbo/blob/main/packages/turbo-utils/src/examples.ts

We could consider opening that check to enforce `.git` suffix only, and then perform the optimized tar download if it's github, otherwise just attempt to grab the repo from the full url.

**Comment by tknickman** [comment]
@NickHackford could you verify that the following works for Github Ent? (I don't have access to an Ent instance myself, so I just want to double check).

Try and visit the following url and ensure it kicks off a download as expected: `https://{GITHUB_ENT_HOST}/{user}/{repo}/archive/{branch}.tar.gz`

**Comment by HacknBashe** [comment]
Hey there. I was out on leave but I'm back now. Sorry for the delay.

I ran the command 

```
yarn dlx turbo gen workspace --copy https://{GITHUB_ENT_HOST}/shared/templates-monorepo.git
```

with the proper host name and it fails when it gets to the part where it starts copying with the error.

```
>>> Invalid URL: "https://{GITHUB_ENT_HOST}/shared/templates-monorepo.git". Only GitHub repositories are supported. Please use a GitHub URL and try again.
```

The turbo version was 1.10.12. I'm not sure if it's failing somewhere else now if I'm doing something wrong.

**Comment by tknickman** [comment]
Hey @NickHackford thanks! No need to test with `turbo` itself yet, I'm just curious if we can download a Github ent repo in the same way as a public repo. 

Can you try just going to:
`https://{GITHUB_ENT_HOST}/{user}/{repo}/archive/{branch}.tar.gz` in your browser and making sure it starts a download?

**Comment by HacknBashe** [comment]
Sorry for the misunderstanding. That url does work and trigger a download of the repo.

**Comment by tknickman** [comment]
No worries at all - awesome! Thanks for confirming.

**Comment by anthonyminyungi** [comment]
Hi, can I get some updates about this?

I was trying to create monorepo template on my company's Github Ent and got the same error message, which brought me here.

**Reply by anthonyshew** [reply]
Please do not add "+1" comments. If you'd like to express your interest, please upvote the top-level.

**Comment by anthonyminyungi** [comment]
Hello @anthonyshew, I noticed that [my comment](https://github.com/vercel/turborepo/discussions/7463#discussioncomment-11761405) was marked as spam, and I was wondering if you could clarify what might have caused this.
When I left the comment, I had no ill intentions and didn’t expect it to be considered spam.
I left a question because I encountered a similar issue while working on a related task and was curious about the progress.
I’ve seen similar update inquiries, or supportive and agreeing comments, in other discussions, so I thought my comment was appropriate. Did I violate any guidelines?
I'd appreciate your guidance if there’s anything I misunderstood or need to improve. I’d like to contribute in the best way possible.
Thank you!

**Reply by anthonyshew** [reply]
We try to have folks not comment with "+1" type comments as they send notifications to anyone trying to keep up with the discussion without providing substantive information. If you'd like to "+1" something, please use the upvote button or a 👍 reaction.

Sorry if "marked as spam" feels too punitive. The options GitHub shows when marking something for hiding don't really have a great option for conveying "please don't add +1 comments". "Spam" is probably too heavy-handed of a description, and folks don't generally have ill intent when +1'ing - but it's the best we have!

**Reply by anthonyshew** [reply]
Mentioned this here, as well, albeit with less words. 😁

https://github.com/vercel/turborepo/discussions/7463#discussioncomment-11837899

**Reply by anthonyminyungi** [reply]
Thank you so much for taking the time to explain this in detail.
This was my first time experiencing something like this, so I was quite surprised and wanted to understand what happened.

I completely understand your explanation and truly agree with it.
I’ll keep this in mind when participating in discussions, both in Turbo and other open-source projects.
Again, I really appreciate your patience and clarification—it was very helpful!
